### PR TITLE
Add font select flags to certificate templates

### PR DESCRIPTION
**Pull request by caseylitton**
@caesar2164 @stvstnfrd PR for adding cme v4 template, changes here are for adding font selection. Has associated PR https://github.com/Stanford-Online/certificate-templates/pull/80.

### CODE REVIEW

File: cert-data.yml [hunk 1]
```diff
@@ -1,6 +1,24 @@
 ---
 # The 4_programmatic certificate renderer uses a special markup language
 #
+# Flags:
+# Font flags need four associated '.ttf' files in Stanford-Online/certificate-templates/template_data/fonts.
+#   The ttf font family files need specific names. One file named
+#       font_name+'-'+BASE+'.ttf',
```
Comment (caesar2164) [inline cert-data.yml]: `BASE` should be `$BASE`

File: cert-data.yml [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+#   where $BASE can be 'Light' or 'Regular'.  And three files named
+#       font_name+'-'+TYPOGRAPHY+'.ttf',
```
Comment (caesar2164) [inline cert-data.yml]: `TYPOGRAPHY` should be `$TYPOGRAPHY`

File: gen_cert.py [hunk 3]
```diff
@@ -48,7 +48,6 @@
 
 reportlab.rl_config.warnOnMissingFontGlyphs = 0
 
-
```
Comment (caesar2164) [inline gen_cert.py:51]: not a relevant change, but seems like a good change...

**Comment by caesar2164** [comment]
this seems like a nicely packaged and documented change, I'd love to see this working!

File: gen_cert.py [hunk 4]
```diff
@@ -1947,16 +1949,15 @@ def _generate_v4_certificate(
...
+        font_string = self.template_font_name + '-' + self.template_font_type
+        addMapping(font_string, 0, 0, font_string)
+        addMapping(font_string, 1, 0, self.template_font_name+'-Italic')
+        addMapping(font_string, 1, 0, self.template_font_name+'-Bold')
+        addMapping(font_string, 1, 1, self.template_font_name+'-BoldItalic')
 
         # These are ordered by preference; cf. font_for_string() above
         self.fontlist = [
-            ('SourceSansPro-Regular', 'SourceSansPro-Regular.ttf', None),
-            ('OpenSans-Light', 'OpenSans-Light.ttf', None),
```
Comment (caesar2164) [inline gen_cert.py:1959]: Looking at this again, we may want keep the OpenSans font even if someone defines a different font as preferred.

Comment (caesar2164) [reply gen_cert.py:1959]: OpenSans has more unicode coverage than most fancier fonts (like SourceSans)

File: cert-data.yml [hunk 5]
```diff
@@ -1,6 +1,24 @@
 ---
 # The 4_programmatic certificate renderer uses a special markup language
 #
+# Flags:
+# Font flags need four associated '.ttf' files in Stanford-Online/certificate-templates/template_data/fonts.
```
Comment (stvstnfrd) [inline cert-data.yml:5]: Should we drop `Stanford-Online` in favor of something institutionally agnostic?

Comment (caesar2164) [reply cert-data.yml:5]: how about: `# Font flags need four associated '.ttf' files in your 'templates/template_data/fonts' directory.` ?

File: gen_cert.py (hunk 4 above)
Comment (caesar2164) [reply gen_cert.py:1959]: nvmd

**Pull request closed without merging**